### DISCUSSION TITLE
Is there a way to change prop name on ThemeProvider

### DISCUSSION THREAD

**Discussion by garridio85** [emotion-js/emotion#2912] [Q&A]
I am trying to use emotion in an existing library but the library already has a prop called theme. Instead of re-writing the existing library I wondered if there is a way to update the name of the prop on the ThemeProvider?

**Comment by Andarist** [comment]
You can always wrap your component and juggle/rename props there.

**Reply by garridio85** [reply]
I tried this but the underlying prop is still coming through as props.theme.foo

this is because the type is set as theme: in the theming.d.ts

**Reply by Andarist** [reply]
You would have to share your code to illustrate the problem because right now I'm not sure what you are struggling with here.
